jakartaee-api/pom.xml

```diff
@@ -87,59 +87,47 @@
             <optional>true</optional>
         </dependency>
 
-        <!-- work around for GLASSFISH-19861  -->
-        <dependency>
-            <groupId>org.glassfish</groupId>
-            <artifactId>jakarta.faces</artifactId>
-            <version>${mojarra.version}</version>
-            <optional>true</optional>
-        </dependency>
-
         <dependency>
             <groupId>jakarta.jms</groupId>
             <artifactId>jakarta.jms-api</artifactId>
             <version>${jakarta.jms-api.version}</version>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>jakarta.activation</groupId>
+            <artifactId>jakarta.activation-api</artifactId>
+            <version>${jakarta.activation-api.version}</version>
+            <optional>true</optional>
+        </dependency>
+
         <!--
             Using com.sun.mail:jakarta.mail instead of
-            jakarta.mail:jakarta.mail-api because javax.mail has a
+            jakarta.mail:jakarta.mail-api because jakarta.mail has a
             dependency on com.sun.mail classes, but the pom does
-            not bring the dependency.
+            not bring in the dependency.
```

Comment:
Good catch, thanks.

Reply:
@bshannon Is it possible to just clean up jakarta.mail so that this dependency on com.sun.mail is properly specified in jakarta.mail pom?  Then, we could clean up this API processing.

Reply:
The point of the API jar file is that you get *just* the APIs, for compiling against,
so you can be sure your code doesn't accidentally depend on an implementation
class.  If the API jar file has a dependency on the implementation jar file, it will
pull in all those classes you were trying to avoid.

There may be some way to avoid this problem, but I don't know how.